Migrate `Investing & Trading..` defichain-wiki.com

Honzan9 opened the pull request:
Migrate/Convert [Investing & Trading..](https://defichain-wiki.com/wiki/Investing_%26_Trading_with_Decentralized_Loans_and_Assets_on_DeFiChain) article from defichain-wiki.com to this new wiki.

- Improve structure with better header levels
- Rework images/diagrams for a better visual style

Honzan9 committed:
Update Investing_Trading.md

Just make the first part with content table and first chapter to test how it will looks like. Need to upload images from the old wiki web - links are directed to "media" folder. Also don't know if links to headers in content table will work. In preview seems not, in desktop VS Code yes.

0ptim committed:
Fixed some Markdown and added sample image

Honzan9 committed:
Add some content, update visual part and add new image

Convert "Part 1" to MD. Added first image updated by figma. I'wonder if it is posible to add a picture with transparent  background? The  script seems to make  all bagrounds solid.  I have to add a light blue bg but without it, it would be much prettier, I think.

0ptim commented:
Hey @Honzan9 
Can you please add the transparent version of the image right below the current image? Then I can try to make it look nice.

Honzan9 committed:
Add Images

Add 3 versions of Image

Honzan9 commented:
Hey @0ptim,
I've just uploaded another 2 versions of the scheme image. first without any background, second with transparent rectangle  background for "center" the image. You maybe noticed, that I chose the colors of the "Wiki  book" which you made the logo of new web. So what do you think? Should I try to make another pictures this way (I enjoy it) and you will improve them if needed? Or should I focus only on the text part?
I hope I will later tonight add other content.

Honzan9 committed:
Add another chapter+image

Add another chapter+image

Honzan9 commented:
Hi again, @0ptim , I added another chapter and try to make a second scheme image for Liquidity mining methods.

0ptim commented:
Hey @Honzan9 
The images actually **are** transparent. I guess the problem you're talking about is that they have a drop shadow as well as a corner radius applied.

## The issue

It's actually me you have to blame for this effect. I added the shadow and radius to all images which are within any article with [CSS](https://www.w3schools.com/css/).

It's this code in `custom.css` which gives this effect.

```css
article img {
  box-shadow: 0px 0px 14px 0px rgb(0 0 0 / 35%);
  border-radius: 6px;
}
```

## Solution 1

We exclude these images somehow and don't apply these effects. This can be done by using this CSS instead. If you decide you want this solution, I can push these changes for you onto your branch.

```css
article img:not([alt*='(TRANSPARENT)']) {
  box-shadow: 0px 0px 14px 0px rgb(0 0 0 / 35%);
  border-radius: 6px;
}
```

Then you can mark your transparent images like so:

```markdown
![basic investment scheme (TRANSPARENT)](./../media/investingtrading_EN_02.png)
```

## Solution 2

We just don't use transparent images. This has also an advantage because you have to always make sure that your transparent images look good in `light` and in `dark` mode. As you already tried, we can do this by just adding the background ourselves in Figma. Maybe we could also experiment a little with modern backgrounds like mesh gradients or alike: https://www.magicpattern.design/tools/mesh-gradients

What do you think we should do?

Btw, I'm positively surprised by your design. Many people would not even consider using the same colors. 💪

Honzan9 commented:
Hey @0ptim , thanks that you make it clear to me! I was quite confused and didn't be sure, if I'm doing wrong the background be transparent or what is going on ;-)  

From solutions I like more the second option. Mainly because of your 100% correct point, that we have to make the web compatible with light and dark mode, It sounds odd to me, but maybe I'm an "oldschool" in this, cause I just never use other than "default" mode of web or app :-D.

About backgrounds, the magic pattern seem to be useful ;-) I'll go throug it.  In the other hand I realy like the "map" background on the homepage and I think it may be the good solution for web "unification", that the images, if they have a background, have the same background as a homepage. What do you think?

I'll try to make some "example(s)" ;-).

FInally I wanna say, that I really appreciate, that you also take the time to "teach" me something. Thanks a lot! I enjoy it ;-)

Honzan9 committed:
update the images

update the images

Honzan9 commented:
Hey @0ptim, I've just try some "magic" with images. Finally I uploaded the version of images, That have bg like a homepage. I try to get through the css to achieve the "switching" the background of the images depening on the dark/light mode = dark/lght map, but sadly I didn't succeed. But I think it may be the way. In light mode it looks good. Just need to solve the "automatic"switch between dark and light.  
Maybe we should also try to find some "DeFiChain theme" bg. Maybe some jellys or coins, preferable in gray = some contures or shadows. Looking forward for your feedback;-)

0ptim commented:
Using the map as a background for this kind of images could be indeed a nice solution. Although we have to make sure that readability does not suffer because something that just looks good but is not usable would be bad.

> Finally I wanna say, that I really appreciate, that you also take the time to "teach" me something. Thanks a lot! I enjoy it ;-)

You're welcome. I think it's important that we help each other out as good as possible. Maybe one day we cross a topic, you can teach me something about too.

---

I just pulled your changes and I think you have some problem with the aspect ratio of the background. It got distorted somehow.

If we plan on further collaborating, we should think about using Figma Teams. I've sent you an invitation over Telegram. I think it would be beneficial to have another call where I can show you quickly on how to work with Figma.

0ptim committed:
Merge branch 'main' into pr/65

0ptim commented:
Updated your branch with latest changes. Don't forget to pull your branch.

Honzan9 committed:
versions of visual style of the article

Some examples to discuss about "How the articles, images and expressions should look like."

Honzan9 commented:
Hi @0ptim,
I pushed an update, beacause I'd like to know your opinion about visual style. I'll try to describe it below.

The Scheme images are updated (also in FIGMA project), back to transparent background. I don't really like previous version with the "fixed" dark-map bg of the image in the "light-mode". So I figured out the way by putting the image into `<div>` and set the background of div depending on the light/dark mode by `custom.css`

The text style is clear and readable, thanks to MD, "highlighting" by `bold` is enough from my point of view. One exception is about "trading examples". Here I try many combinations and I simply cannot decide which is "the right one" from following versions (in order which can you see on the page)

 1) simple text like other parts of the article, only important "bold"
 2) `<div>` with map background, whole text bold
 3) `<div>` with map background,  only important "bold"
 4) `<div>` with map background,  only important "bold" + the expresions with `"code style"`

I've also looked on the KaTex and I think it is powerful tool, but for our "math operations" maybe overkilled ;-)

So I'd be glad, if you will find some time to check it and give me a feedback.

Thanks.

0ptim commented:
Hey @Honzan9 
Nice that you tried so many things and experimented with CSS/HTML. This way you will learn a lot about the technologies.
It's indeed difficult to decide which version fits best. But the following comes to mind:

If we want to use the map background for text, we really need to make sure, that the text is very easily readable. So for how it is now, I think the version `2` with the bold font is the best.
The disadvantage there is, that we commit to always have the text in bold. So we lose the common formatting logic which allows us to highlight specific parts with bold. We can go with this option if you don't want to further experiment and just want to go on with the conversion. We can always come back to this topic in the future.

An option would be to have a different version of the map background. Maybe just the `dull` version of the map from Figma. Or we create an **even more** dull version for these scenarios. This way the contrast between the light and dark parts of the map are even less, so the text is very easy to read, even without the font being bold. We could also experiment with making a map version where we blur the image.

Another thing would be to use the [Admonitions](https://docusaurus.io/docs/markdown-features/admonitions) from Docusaurus. But if you don't like these, you don't have to use them!

Let me know what you think.

PS: I'm not sure what's causing this?
![image](https://user-images.githubusercontent.com/25133399/215414288-e511f59a-55d5-4069-9253-6e140f8f7933.png)

Honzan9 commented:
Hey @0ptim,
thanks for you thoughts!  "All bold" is not the way IMO, I fully agree with your point about highlighting.

I've been also thinking about make another "lighter" or "blur" version of the map, but I'm afraid, that maybe it will be to much to have more types of background on one page. So If we will go this way, we should make one version that fits also like bg for images. I'll try to make some drafts ;-)

About Admonitions, I've read the article on Docusaurus web already, but I haven't enough courage to try making "custom" version :-) At least we need to change a color to our pink-blue scheme, it is even possible? If yes, I think it would be the way for the "examples", but I'll need your help here ;-).

The "strange borderline" of the image is the `article img: box-shadow` from `custom.css`, I think. TBH I've just wanna try if you will notice it ;-) like some kind of "user testing".  I think we can "remove" it for the dark mode, in the light it looks good IMO.

For now I can continue with the content review, there is a lot of  stuff and I'm not sure, that is all up to date, so I can also improve my DeFiChain knowledge by going through this ;-). 

So we have time to find the best style meanwhile.

0ptim commented:
@Honzan9 Maybe we could just use a very “dull” version of the map for everything (Homepage/text-background/image-background).

Actually, I haven't even considered using custom admonitions. I was just talking about using normal admonitions. But maybe we could also change the color of an existing admonition, i.e., change `info` from the blue color to a more DeFiChain-ish color. I think this would not be very difficult.

Honzan9 committed:
Update content, settle the style

Images with map BG, examples with Admonitions, content review of Part 1 & 2 done.

Honzan9 commented:
Hey @0ptim, I was ill last week and therefore no active as much as I want. Now I pushed another update. In the end I realize that the Admonitions for "examples" fits quite good to my vision. In this version the Images have the mapbackground, and the examples are framed by blue "info" admonition.  About the content review I think I'm done with chapters 1 and 2.
Give me a sign if you like my "final" style or what to change. Thanks.

0ptim committed:
Merge branch 'main' into pr/65

0ptim commented:
Hi @Honzan9 
Good to hear you feel better now!
I think it looks great. 👍🏼 I have made you a message on Telegram about working with Git.

0ptim committed:
Run prettier

Honzan9 committed:
Final version for review

The full content updated in new style (except last chapter about arbitrage, which is not completely up to date)

Honzan9 commented:
Hi @0ptim,
I think my current commit should be the first "final" version for review. I updated whole content except the last chapter about arbitrage, which IMO is not up to date and because of recent status about DUSD depeg, high stab. fee etc it is quite difficult to describe all aspects of arbitrage and TBH I'm not sure if it is even valuable for common wiki user to know specific arbitrage mechanics on DeFiChain.

I'll be glad if you will check the content (generally). As I'm looking into it long, it easy to overlook some inaccuracies, typos etc.

If it will be ok, I will switch to the page about The DEX.

Loking forward for your feedback!

0ptim commented:
Hey @Honzan9 

Good hearing from you! 😊 Great work!

### Just for discussion
1. Did you go for a different design of these on purpose?
![image](https://user-images.githubusercontent.com/25133399/219380323-e4f076f1-266a-409b-996f-75791ea3f78e.png)
2. Do you think it's necessary to include `Part X: `? The titles are kind of long already without it.
![image](https://user-images.githubusercontent.com/25133399/219380556-a4596d8d-a4b1-4443-b109-447ab639c7b2.png)
3. This can go: `This page was not fully converted yet to the new wiki format. Coming soon!`
4. This can go: `The source of this page can be found here: https://defichain-wiki.com/wiki/Investing_%26_Trading_with_Decentralized_Loans_and_Assets_on_DeFiChain`
5. I think this shadow is a little weird 😄
![image](https://user-images.githubusercontent.com/25133399/219381169-74f2495d-b26e-44b0-804b-4a76d5e1a876.png)
6. Just an idea: Would it be nicer to split this page into 3 different pages and add a category like `Investing and Trading`?
![image](https://user-images.githubusercontent.com/25133399/219381697-022674cd-ab73-44d4-b653-317612683fb3.png)

Honzan9 commented:
Hi @0ptim,

thanks for your thoughts. My answers below:

1. About "bubbles style": Yes, I've  just tried to make it little different but still simillar for tables. I cannot decide which way is best now. Make it same as the schemes (linear gradient) or all as the tables (radial gradient) or leave it with both version. What do you think?

2. About "long" headers: Of course we can shorten them or divide the page to more pages - as point 6 propose.

3.+4. Ok 

5. Weird shadow: Thanks for the point, I forgot on it - will solve in css.

6. About page split: For my point of view are both ways possible, spliting to smaller parts maybe fits better to "modern" style of presentation.  May I try to split the page, or will be better if you will do?

Thanks again ;-)

I'm having a "rush" time now, especially because of work, but I'm glad I can participate ;-)

H.

0ptim commented:
Hey @Honzan9 
1. You can decide. Just try to make it easily adjustable in Figma. Then we are good if we want to change something. 😊
2. I let you decide.
3. :)
4. :)
5. Nice
6. I think you can do it. Just decide on yourself. It was just an idea.

You're welcome!

Honzan9 committed:
Merge branch 'main' into pr/65

Honzan9 committed:
Last check before merge

Proposed changes implemented with exception about page split.

Honzan9 commented:
Hey @0ptim, I think we can merge. I thik that I've solved all the points except the page split. I'll return to it when I'll be working on the "chapter reorganization" in the new branch ;-).

0ptim commented on src/css/input.css:
```diff
@@ -118,6 +123,36 @@ article a[href*="//"]:not([href*="defichainwiki.com"])::after
   content: "" !important;
 }
 
+.formula_bubble {
```
Where is this needed?

Honzan9 replied:
Nowhere, it was used in one of the version of solving backgrounds for numeric examples.  It should be removed.

0ptim commented on src/css/input.css:
```diff
@@ -118,6 +123,36 @@ article a[href*="//"]:not([href*="defichainwiki.com"])::after
+  padding: 0px 0px 0px 0px;
+}
+
+.formula_map {
```
Where is this needed?

Honzan9 replied:
Nowhere, it was used in one of the version of solving backgrounds for numeric examples.  It should be removed.

0ptim commented on src/css/input.css:
```diff
@@ -118,6 +123,36 @@ article a[href*="//"]:not([href*="defichainwiki.com"])::after
+  padding: 30px 30px 10px 30px;
+}
+
+.formula_map_bold {
```
Where is this needed?

Honzan9 replied:
Nowhere, it was used in one of the version of solving backgrounds for numeric examples.  It should be removed.

0ptim commented:
Did you want to also add `background-size: cover;` for all map background CSS classes? This way, the map is not zoomed in.

Honzan9 committed:
CSS review

Removed unused definitions and add the 'background-size: cover;'

Honzan9 commented:
Hey @0ptim, thanks for your review. I removed unused definitions in CSS file and added the 'background-size: cover;' parameter, so BG looks far more better now.
